synapse/storage/databases/main/keys.py

```diff
@@ -236,7 +236,7 @@ def _get_keys(txn: Cursor, batch: Tuple[Tuple[str, str], ...]) -> None:
 
                 # The entire signed JSON response is stored in server_keys_json,
                 # fetch out the bits needed.
-                key_json = json.loads(key_json_bytes)
+                key_json = json.loads(bytes(key_json_bytes))
```

Comment:
Bit confused by this.

1. Doesn't json.loads accept [strings or bytes](https://docs.python.org/3/library/json.html#json.loads)?
2. The commit says this fixes postgres. But the column is `bytea` in postgres; I'd expect this gets loaded as a `bytes` by psycopg2. So I don't understand why this commit makes a difference to postgres?

Reply:
This converts from a `memoryview` (which is what postgres/psycopg2 returns for a `bytea` file) to `bytes`, we do this in a few other spots in the file.

Reply:
Maybe -- I don't see a bug for it, but you might be able to automatically do it in the decoder. 🤷